L1Trigger/Phase2L1ParticleFlow/plugins/L1CTJetFileWriter.cc

```diff
@@ -87,13 +130,13 @@ void L1CTJetFileWriter::endJob() {
   fileWriterOutputToGT_.flush();
 }
 
-std::vector<ap_uint<64>> L1CTJetFileWriter::encodeJets(const std::vector<l1t::PFJet> jets) {
+std::vector<ap_uint<64>> L1CTJetFileWriter::encodeJets(const std::vector<l1t::PFJet> jets, const unsigned nJets) {
   std::vector<ap_uint<64>> jet_words;
-  for (unsigned i = 0; i < nJets_; i++) {
+  for (unsigned i = 0; i < nJets; i++) {
     l1t::PFJet j;
     if (i < jets.size()) {
       j = jets.at(i);
-    } else {  // pad up to nJets_ with null jets
+    } else {  // pad up to nJets with null jets
```

Comment:
Here this piece was already wrong to begin with since the new definition of `PFJet j` in the `else` branch doesn't really replace the one defined outside the if.

What you want is
```cpp   
    l1t::PFJet j(0, 0, 0, 0, 0, 0); // initialize as null
    if (i < jets.size()) {
      j = jets.at(i);
    }
```

Two other alternatives:
 * upstream you could do `jets.resize(nJets, l1t::PFJet(0, 0, 0, 0, 0, 0));`  (not here since it's const)
 * stop the loop if `i >= jets.size();` and then just do `jet_words.resize(2*nJets, ap_uint<64>(0))`